spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/spring/AbstractArmeriaAutoConfiguration.java

```diff
@@ -69,16 +69,16 @@ public Server armeriaServer(
             ArmeriaSettings armeriaSettings,
             InternalServices internalService,
             Optional<MeterRegistry> meterRegistry,
-            Optional<List<MetricCollectingServiceConfigurator>> metricCollectingServiceConfigurators,
+            ObjectProvider<MetricCollectingServiceConfigurator> metricCollectingServiceConfigurators,
             Optional<MeterIdPrefixFunction> meterIdPrefixFunction,
-            Optional<List<ArmeriaServerConfigurator>> armeriaServerConfigurators,
-            Optional<List<Consumer<ServerBuilder>>> armeriaServerBuilderConsumers,
-            Optional<List<DependencyInjector>> dependencyInjectors,
-            Optional<List<ServerErrorHandler>> serverErrorHandlers,
+            ObjectProvider<ArmeriaServerConfigurator> armeriaServerConfigurators,
+            ObjectProvider<Consumer<ServerBuilder>> armeriaServerBuilderConsumers,
+            ObjectProvider<DependencyInjector> dependencyInjectors,
+            ObjectProvider<ServerErrorHandler> serverErrorHandlers,
             BeanFactory beanFactory) {
 
-        if (!armeriaServerConfigurators.isPresent() &&
-            !armeriaServerBuilderConsumers.isPresent()) {
+        if (!armeriaServerConfigurators.stream().findAny().isPresent() &&
+            !armeriaServerBuilderConsumers.stream().findAny().isPresent()) {
```

Comment:
Optional) Not strong on this, but intelliJ is complaining and generally inverse conditions are slightly harder to reason about (although I think this case is simple enough)

```suggestion
        if (armeriaServerConfigurators.stream().findAny().isEmpty() &&
            armeriaServerBuilderConsumers.stream().findAny().isEmpty()) {
```

Reply:
Thanks for your suggestion and yes I have used `isEmpty()` at first, but local test failed due to this, not sure the exact reason because error message is garbled characters.

Reply:
`Optional.isEmpty()` is introduced in Java 11 but our baseline of Java is 8.